### PR TITLE
fix: Single-packet RTX in Initial/Handshake + RTX handshake ACKs

### PR DESCRIPTION
**Pull request by larseggert**
With this change, we're now only sending one packet on PTO in the Initial and Handshake packet number spaces, for both server and client. Before, the server was allowed to send two and the client was allowed to send two during Handshake, which meant that if one packet was lost, we would RTX the lost packet and a (useless) empty PING. The server would then hit the amplification limit. With this change, we only RTX the lost packet, making the Handshake more robust under loss.

A second change here is to immediately force an ACK during the Handshake when a previous ACK was lost. Before this, the original Initial form the server contained an ACK, but any RTX did not, because we stopped the ACK timer when generating the first ACK and never restarting it.

Both these changes reduce `L1` and `C1` interop failures.

### CODE REVIEW

File: neqo-transport/src/connection/tests/handshake.rs
Comment (mxinden) [inline neqo-transport/src/connection/tests/handshake.rs:1245]:
```suggestion
        let si = server.process(&ci.take(), now).dgram().unwrap();
        assert_eq!(si.len(), server.plpmtu());
        assert_eq!(
            server.stats().frame_tx,
            FrameStats {
                crypto: i * 2,
                ack: i,
                all: i * 3,
                ..Default::default()
            }
        );
```

Using `take()` is a bit simpler and concise. Haven't tried to compile it. Feel free to ignore.

File: neqo-transport/src/tracking.rs
Comment (mxinden) [inline neqo-transport/src/tracking.rs:606]: I believe the doc comment below also needs updating:

https://github.com/mozilla/neqo/blob/0eb9174d7a67b250607f0c3ea056fe056bcf91f5/neqo-transport/src/tracking.rs#L420-L424

**Review by mxinden — COMMENTED** [review]
For what my review is worth, this makes sense to me. That said, I suggest waiting for another review, given the subtle nature of the change.

**Comment by larseggert** [comment]
That last change seems to really have helped with `L1` and `C1`. I think this is ready for review.

File: neqo-transport/src/recovery/mod.rs
Comment (larseggert) [inline neqo-transport/src/recovery/mod.rs:446]:
```suggestion
            // For the Initial and Handshake spaces, we only send one packet on PTO. This avoids
```

File: neqo-transport/src/tracking.rs
Comment (larseggert) [inline neqo-transport/src/tracking.rs:302]:
```suggestion
            } else {
                // ACK more aggressively
```

**Review by mxinden — APPROVED** [review]

**Pull request merged**
